rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,124 @@
+
+The `AS` parameter name, if supported, will create an alias for the dynamic metric which will be used in the schema of the response as well as in the having clause of the request.
+
+Example:
+
+```
+?metrics=users(filter,dim1|id-in['a','b'],AS=DimOneInAOrB)&having=DimOneInAOrB-gt[50]
+```
+
+This would create a variation of the standard `users` aggregation which only counted users from records where the dimension `dim1` had values "a" or "b", and would display result rows only which for which this metric had at least 50 users.
+
+Parameterized Metric Metadata
+=============================
+
+The metrics endpoint (and the corresponding serialization in the fullView format on the tables endpoint) will now contain an optional field for parameters.
+
+```
+{ 
+    name:"grossRevenue",
+    type:"money"
+    parameters: {
+        currency: {  
+            type: dimension, 
+            dimensionName: "currencyUnits",
+            defaultValue:"billing" 
+        }
+    }
+}
+```
+
+Parameters will be a JSON object whose keys are parameter names supported on this metric.
+
+The values of a parameter entry will contain a mandatory field 'type'.
+
+In this example, the `dimension` supports fields `dimensionName` and `defaultValue`.  Dimension name is the name of a dimension whose values are legal for this parameter.  
+
+The `defaultValue` field is an optimization for UI clients to provide a 'recommended' value for the parameter, if appropriate.  `defaultValue`, if used, is recommended to be equivalent to not supplying a value for this parameter at all.
+
+For conciseness, the default output format for the metric endpoint will NOT include reserved word system-wide parameters.  A query parameter for verbose expansion may be subsequently defined. 
+
+Parameter Types
+---------------
+
+Recommended types to be implemented:
+
+`identifier`: The parameter value will be parsed as an identifier string.
+
+`filter`: The parameter accepts values parsed as the entries in the filters query parameter.
+
+`dimension`: The parameter accepts values from a dimension.
+
+Proposed future type extensions:
+
+`enum`: The parameter accepts values from an included list of values.
+
+`number`: The parameter accepts simple (signed) numeric values.
+
+`date`: The parameter accepts values that parse as the DateTime part of the interval used within dateTime query parameter.
+
+`interval`: The parameter accepts intervals as used within dateTime query parameter.
+
+Reserved Parameter Types
+------------------------
+
+The `filter` reserved parameter should have an implicit type of `filter`
+
+The `AS` reserved parameter should have an implicit type of `identifier`
```

Comment:
Can we have a section on error handling? Incorrect param name, bad value (type), syntax error, etc.